ovh/resource_domain_name_servers.go

```diff
@@ -0,0 +1,220 @@
+	d *schema.ResourceData,
+	meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	d.SetId(givenId)
+	d.Set("domain", d.Id())
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func resourceOvhDomainNameServers() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOvhDomainNameServersCreate,
+		Read:   resourceOvhDomainNameServersRead,
+		Update: resourceOvhDomainNameServersCreate, // Update is the same as create
+		Delete: resourceOvhDomainNameServersDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceOvhDomainNameServersImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"servers": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ip": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "",
+						},
+					},
+				},
+				MinItems: 1,
+			},
+		},
+	}
+}
+
+func resourceOvhDomainNameServersCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	serviceName := d.Get("domain").(string)
+
+	// Servers to put
+	servers := &OvhDomainNameServerUpdate{
+		NameServers: make([]OvhDomainNameServer, 0),
+	}
+	// Loop due to rename of the field in the API
+	for _, server := range d.Get("servers").([]interface{}) {
+		s := server.(map[string]interface{})
+		servers.NameServers = append(servers.NameServers, OvhDomainNameServer{
+			Host: s["host"].(string),
+			Ip:   s["ip"].(string),
+		})
+	}
+	log.Printf("[DEBUG] OVH Record create configuration: %#v", servers)
+	err := config.OVHClient.Post(fmt.Sprintf("/domain/%s/nameServers/update", url.PathEscape(serviceName)), servers, nil)
+
+	if err != nil {
+		return fmt.Errorf("failed to register OVH Nameservers: %s", err)
+	}
+
+	d.SetId(serviceName)
+
+	return resourceOvhDomainNameServersRead(d, meta)
+}
+
+func resourceOvhDomainNameServersRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	record, err := ovhDomainNameServers(config.OVHClient, d)
+
+	if err != nil {
+		return err
+	}
+
+	if record == nil {
+		return fmt.Errorf("domain %v has been deleted", d.Id())
+	}
+
+	d.SetId(record.Id)
+	d.Set("domain", record.Domain)
+	d.Set("servers", flattenOvhDomainNameServers(record.Servers))
+
+	return nil
+}
+
+func flattenOvhDomainNameServers(servers []OvhDomainNameServer) []interface{} {
+	result := make([]interface{}, 0)
+	for _, server := range servers {
+		result = append(result, map[string]interface{}{
+			"host": server.Host,
+			"ip":   server.Ip,
+		})
+	}
+	return result
+}
+
+func resourceOvhDomainNameServersDelete(d *schema.ResourceData, meta interface{}) error {
+	// Note that NameServers can not be deleted, only updated
```

Comment:
when deleting, shouldn't we call `/domain/xxx/nameServers/update` with an empty array ? If this doesn't work, you should call `DELETE /domain/{serviceName}/nameServer/{id}` on each nameServer to clean them.